src/test/java/org/dom4j/io/StaxTest.java

```diff
@@ -50,7 +51,7 @@ public void testEncoding() {
         try {
             File file = getFile("/xml/russArticle.xml");
             STAXEventReader xmlReader = new STAXEventReader();
-            Document doc = xmlReader.readDocument(new FileReader(file));
+            Document doc = xmlReader.readDocument(new FileInputStream(file));
```

Comment:
Using a reader means the underlying encoding is not read has to use input stream as per https://stackoverflow.com/questions/3482494/howto-let-the-sax-parser-determine-the-encoding-from-the-xml-declaration